mad/mongo-adapter.go

```diff
@@ -0,0 +1,76 @@
+package mad
```

Comment:
Any reason not to test this?

Reply:
It is a thin wrapper around mongo driver, don't think it calls for unit tests. 
Would consider doing e2e test at a later stage.